# Bowtie2 Generate Genome Index

This capsule generates the FM-suffix genome index for Bowtie2 using default parameters. 

## Input Data

**Genome Fasta File(s)**

Nucleotide sequences to generate the index from. Any \*.fa\*, \*.fna\* file(s) in **data** directory will be used. 

If there are multiple reference sequences with identical suffixes (i.e., .fa), sequences will be concatenated into a single file before indexing. Please ensure that you are using an identical suffix for all your sequences. 

## App Panel parameters

Data : Hg38 Reference
- The hg38 reference is attached as default.
    - If you should decide to change it, please substitute this dataset with your own reference sequence. 

Index File Name
- Output name for index file. If not specified, will be based on the first discovered input fasta name.

## Output

A genome index generated by Bowtie2, including the following 6 files:

- \*.1.bt2
- \*.2.bt2
- \*.3.bt2
- \*.4.bt2
- \*.rev.1.bt2
- \*.rev.2.bt2

## Source

https://github.com/BenLangmead/bowtie2

## Cite

Langmead B, Wilks C, Antonescu V, Charles R. Scaling read aligners to hundreds of threads on general-purpose processors. Bioinformatics. 2018 Jul 18. doi: 10.1093/bioinformatics/bty648.

Langmead B, Salzberg SL. Fast gapped-read alignment with Bowtie 2. Nature Methods. 2012 Mar 4;9(4):357-9. doi: 10.1038/nmeth.1923
